src/ser/value.rs

```diff
@@ -158,13 +158,13 @@ impl<'key, 'target, Target> Serializer
     }
 
     fn serialize_none(&mut self) -> Result<(), Error> {
-        Err(Error::unsupported_value())
+        Ok(())
     }
 
     fn serialize_some<T>(&mut self, _value: T) -> Result<(), Error>
         where T: Serialize
     {
-        Err(Error::unsupported_value())
+        _value.serialize(self)
```

Comment:
Since this is no longer an unused variable, it shouldn't be prefixed with an `_`.